hasher.go

```diff
@@ -23,7 +23,18 @@ var (
 	ErrInvalidNodeNamespaceOrder = errors.New("invalid NMT node namespace order")
 )
 
-type Hasher struct {
+// NmtHasher describes the interface nmts use to hash leafs and nodes.
```

Comment:
[nit] consistency between interface name and name referenced in comment. Since there is an NmtHasher type below, I wonder if this comment meant to be above line 37? Alternatively: 

```suggestion
// Hasher describes the interface NMT uses to hash leafs and nodes.
```

Reply:
sorry, forgot to mention that this was fixed in https://github.com/celestiaorg/nmt/pull/211/commits/2f7f25ec5e01ec8cb9d24509de1e32b510dda70d